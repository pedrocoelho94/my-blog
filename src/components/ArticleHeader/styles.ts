import styled, { css } from 'styled-components'
import * as HeadingStyles from 'components/Heading/styles'

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles.Title} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};
      font-size: ${theme.font.sizes.large};
    }
  `}
`

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: calc(${theme.font.sizes.medium} * 0.9);
  `}
`

export const Cover = styled.img`
  ${({ theme }) => css`
    max-width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`
