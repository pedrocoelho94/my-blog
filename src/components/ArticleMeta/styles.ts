import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span:not(:last-child):after {
      content: ', ';
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 0.2s ease-in-out;

      &:hover {
        filter: brightness(0.5);
      }
    }
  `}
`
