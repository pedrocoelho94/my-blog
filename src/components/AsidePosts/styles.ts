import styled, { css } from 'styled-components'

export const Wrapper = styled.div``

export const Content = styled.a`
  ${({ theme }) => css`
    display: flex;
    gap: 1rem;
    height: 100%;
    cursor: pointer;
    text-decoration: none;
    margin: ${theme.spacings.medium} 0;
  `}
`

export const Cover = styled.img`
  border-radius: 0.8rem;
  width: 10rem;
  height: 10rem;
  object-fit: cover;
`

export const Title = styled.h4`
  ${({ theme }) => css`
    font-size: 1.5rem;
    height: 100%;
    margin: 0;
    color: ${theme.colors.darker};

    &:hover {
      text-decoration: underline;
    }
  `}
`
