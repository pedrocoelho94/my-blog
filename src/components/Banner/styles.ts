import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  position: relative;
`

export const Content = styled.div`
  filter: brightness(0.95);
  transition: filter 0.2s ease-in-out;

  &:hover {
    filter: brightness(1.2);
  }
`

type imgProps = {
  src: string
}

export const Image = styled.div<imgProps>`
  ${({ src }) => css`
    width: 100%;
    height: 30rem;
    background-image: url(${src});
    background-position: center center;
    background-size: cover;

    ${media.greaterThan('small')`
      height: 40rem;
    `}

    ${media.greaterThan('medium')`
      height: 45rem;
    `}
  `}
`
export const Caption = styled.div`
  ${({ theme }) => css`
    width: 100%;
    padding: ${theme.spacings.medium};
    background-color: rgba(0, 0, 0, 0.4);
    position: absolute;
    bottom: 0;
    left: 0;
  `}
`

export const Title = styled.h2`
  ${({ theme }) => css`
    margin-top: 0;
    color: ${theme.colors.white};

    ${media.lessThan('small')`
      font-size: 2rem;
      margin-bottom: 0;
    `}
  `}
`

export const Subtitle = styled.h3`
  ${({ theme }) => css`
    font-weight: inherit;
    margin-bottom: 0;
    color: ${theme.colors.white};

    ${media.lessThan('small')`
      display: none
    `}
  `}
`
