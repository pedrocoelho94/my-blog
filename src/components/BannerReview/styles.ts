import styled, { css } from 'styled-components'

export const Wrapper = styled.div``

export const Poster = styled.div`
  position: relative;
  width: 100%;
  max-width: 25rem;
  height: 100%;
  max-height: 35rem;
  transition: all 0.2s ease-in-out;
  overflow: hidden;
  border-radius: 5px;
`

export const Image = styled.img`
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-radius: 5px;
  transition: all 0.2s ease-in-out;
  transform: scale(1.05);

  &:hover {
    transform: scale(1.1);
  }
`

type RatingProps = {
  rating: number
}

const RatingModifiers = {
  good: () => `
    background-color: #01d200;
  `,
  ok: () => `
    background-color: #ffb300;
  `,
  bad: () => `
    background-color: #ff7e00;
  `,
  terrible: () => `
    background-color: #df0700;
  `
}

export const Rating = styled.div<RatingProps>`
  ${({ theme, rating }) => css`
    position: absolute;
    bottom: 0;
    right: 0;
    z-index: 5;
    color: ${theme.colors.white};
    font-size: 1.8rem;
    font-weight: bold;
    width: 3.5rem;
    height: 3.5rem;
    display: flex;
    border-radius: 5px;

    span {
      width: 100%;
      text-align: center;
      justify-self: center;
      align-self: center;
    }

    ${rating >= 7.5 && RatingModifiers.good()}
    ${rating >= 5 && rating < 7.5 && RatingModifiers.ok()}
    ${rating >= 2.5 && rating < 5 && RatingModifiers.bad()}
    ${rating < 2.5 && RatingModifiers.terrible()}
  `}
`
