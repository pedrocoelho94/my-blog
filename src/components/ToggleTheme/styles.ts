import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  position: fixed;
  top: 2rem;
  right: 2rem;
  z-index: 2;
`
export const Label = styled.label`
  position: relative;
  display: inline-block;
  width: 5rem;
  height: 2.5rem;
  line-height: 0;
  font-size: 0;
  overflow: hidden;
  color: rgba(0, 0, 0, 0);
`
export const Input = styled.input`
  ${({ theme }) => css`
    appearance: none;
    opacity: 0;
    width: 0;
    height: 0;

    &:checked + ${Slider} {
      background: ${theme.colors.secondary};
      border: 1px solid ${theme.colors.secondary};
    }

    &:focus + ${Slider} {
      border: 1px solid ${theme.colors.secondary};
    }

    &:checked&:focus + ${Slider} {
      border: 1px solid ${theme.colors.primary};
    }

    &:checked + ${Slider}:before {
      transform: translateX(2rem);
      background: ${theme.colors.primary};
    }
  `}
`
export const Slider = styled.span`
  ${({ theme }) => css`
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: ${theme.colors.darkerGray};
    transition: all 0.3s ease-in-out;
    border-radius: 2rem;
    border: 1px solid ${theme.colors.darkerGray};

    &:before {
      content: '';
      position: absolute;
      height: 1.9rem;
      width: 1.9rem;
      left: 0.4rem;
      bottom: 0.2rem;
      background: ${theme.colors.white};
      border-radius: 50%;
      transition: all 0.3s ease-in-out;
    }
  `}
`
