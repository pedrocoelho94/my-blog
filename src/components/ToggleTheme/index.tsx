import { BlogThemeContext } from 'context/BlogThemeContext'
import { useState, useContext, useEffect } from 'react'
import * as S from './styles'

const ToggleTheme = () => {
  const { setTheme } = useContext(BlogThemeContext)
  const [isChecked, setIsChecked] = useState(false)

  useEffect(() => {
    const localTheme = localStorage.getItem('theme')
    if (!localTheme) return
    const newTheme = JSON.parse(localTheme)

    if (newTheme.name === 'inverted') {
      setIsChecked(true)
    }
  }, [])

  useEffect(() => {
    setTheme?.(isChecked ? 'inverted' : 'default')
  }, [isChecked, setTheme])

  const handleChange = () => {
    setIsChecked((isChecked) => !isChecked)
    setTheme?.(isChecked ? 'inverted' : 'default')
  }

  return (
    <S.Wrapper>
      <S.Label>
        Toggle light and dark modes
        <S.Input
          type="checkbox"
          value="Dark mode active"
          onChange={handleChange}
          checked={isChecked}
        />
        <S.Slider></S.Slider>
      </S.Label>
    </S.Wrapper>
  )
}

export default ToggleTheme
