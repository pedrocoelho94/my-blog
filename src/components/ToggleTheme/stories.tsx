import { Story, Meta } from '@storybook/react'
import ToggleTheme from '.'

export default {
  title: 'ToggleTheme',
  component: ToggleTheme
} as Meta

export const Default: Story = () => <ToggleTheme />
