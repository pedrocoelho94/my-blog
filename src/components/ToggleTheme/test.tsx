import { renderWithTheme } from 'styles/renderWithTheme'

import ToggleTheme from '.'

describe('<ToggleTheme />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<ToggleTheme />)

    expect(container).toMatchSnapshot()
  })
})
