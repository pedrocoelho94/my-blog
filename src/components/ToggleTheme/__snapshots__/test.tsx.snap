// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleTheme /> should render the heading 1`] = `
.c0 {
  position: fixed;
  top: 2rem;
  right: 2rem;
  z-index: 2;
}

.c1 {
  position: relative;
  display: inline-block;
  width: 5rem;
  height: 2.5rem;
  line-height: 0;
  font-size: 0;
  overflow: hidden;
  color: rgba(0,0,0,0);
}

.c3 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  opacity: 0;
  width: 0;
  height: 0;
}

.c3:checked + .c4 {
  background: #dc143c;
  border: 1px solid #dc143c;
}

.c3:focus + .c4 {
  border: 1px solid #dc143c;
}

.c3:checked.c2:focus + .c4 {
  border: 1px solid #000000;
}

.c3:checked + .c4:before {
  -webkit-transform: translateX(2rem);
  -ms-transform: translateX(2rem);
  transform: translateX(2rem);
  background: #000000;
}

.c5 {
  position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: #AAAAAA;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  border-radius: 2rem;
  border: 1px solid #AAAAAA;
}

.c5:before {
  content: '';
  position: absolute;
  height: 1.9rem;
  width: 1.9rem;
  left: 0.4rem;
  bottom: 0.2rem;
  background: #FFFFFF;
  border-radius: 50%;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1"
    >
      Toggle light and dark modes
      <input
        class="c2 c3"
        type="checkbox"
        value="Dark mode active"
      />
      <span
        class="c4 c5"
      />
    </label>
  </div>
</div>
`
